### PR TITLE
Support arm builds

### PR DESCRIPTION
**Pull request by rob-baily**
# PR Details
## Description

Allow for architecture to be passed in for c++ build workflow.
NOTE: When using another architecture on Ubuntu the apt-get update command will sometimes fail because the package files for the architecture do not exist on the repo.  For this we always consider apt-get update to succeed as any issues should be caught in the next steps.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by rob-baily** [comment]
Related to the build image updates at https://github.com/usdot-fhwa-stol/carma-builds/pull/14

File: setup-deb-s3/action.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ runs:
         fi
 
         if ! which lsb_release; then
-          $sudocmd apt-get update
+          $sudocmd apt-get update && true
```
Comment (codygarver) [inline setup-deb-s3/action.yml:25]: I think you meant `|| true`

Also I'd like to see an example where this fails

**Review by codygarver — CHANGES_REQUESTED** [review]

File: setup-deb-s3/action.yml (hunk 1 above)
Comment (rob-baily) [reply setup-deb-s3/action.yml]: You can see where it fails here: https://github.com/usdot-fhwa-stol/snmp-client/actions/runs/4285308321/jobs/7463443650  It has something to do with using the arm-hf architecture and what the repos return for this.  Strangely it seems the && does work which I was not expecting after your comment so I will try it with || now.

Comment (rob-baily) [reply setup-deb-s3/action.yml]: Changed it to || and it works as shown here https://github.com/usdot-fhwa-stol/snmp-client/actions/runs/4285355481/jobs/7463552983

**Pull request merged**